dataset/src/test/scala/frameless/functions/AggregateFunctionsTests.scala

```diff
@@ -197,6 +208,18 @@ class AggregateFunctionsTests extends TypedDatasetSuite {
     check(forAll(prop[String] _))
   }
 
+  test("max with follow up multiplication") {
+    def prop(xs: List[Long]): Prop = {
+      val dataset = TypedDataset.create(xs.map(X1(_)))
+      val A = dataset.col[Long]('a)
+      val datasetMax = dataset.agg(max(A) * 2).collect().run().headOption
```

Comment:
How confident are you that *all* operations are supported on aggregated columns? Given what you've found with orderBy I wouldn't be surprised if only a subset was, so maybe we should be a bit more exhaustive. If there was a way to do that without duplicating all the column tests, that would be ideal.

Reply:
pretty confident. All the columns inside an `agg()` after you applied an aggregation method, then they are treated as regular columns and you they support all ops. This is a bit different from orderBy, where the selected columns are not really projections; just columns you use to order the data. On the other hand, with `select()` and `agg()`, columns included there are the columns on the resulting dataframe (its schema), and I found those to be consistent.